Description: Additional changes to work with the Debian package of singular
Author: Tobias Hansen <thansen@debian.org>
Author: Ximin Luo <infinity0@debian.org>
Forwarded: not-needed
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/sage/src/bin/sage
+++ b/sage/src/bin/sage
@@ -467,7 +467,7 @@
 
 if [ "$1" = '-singular' -o "$1" = '--singular' ]; then
     shift
-    exec singular "$@"
+    exec Singular "$@"
 fi
 
 if [ "$1" = '-sqlite3' -o "$1" = '--sqlite3' ]; then
--- a/sage/src/sage/interfaces/singular.py
+++ b/sage/src/sage/interfaces/singular.py
@@ -2272,7 +2272,7 @@
     nodes.clear()
     node_names.clear()
 
-    singular_docdir = os.environ["SAGE_LOCAL"]+"/share/singular/"
+    singular_docdir = "/usr/share/doc/singular-doc/"
 
     new_node = re.compile("File: singular\.hlp,  Node: ([^,]*),.*")
     new_lookup = re.compile("\* ([^:]*):*([^.]*)\..*")
--- a/sage/src/sage/libs/singular/singular.pyx
+++ b/sage/src/sage/libs/singular/singular.pyx
@@ -24,6 +24,7 @@
     long INT_MIN
 
 import os
+import sysconfig
 
 from libc.stdint cimport int64_t
 from sage.libs.singular.decl cimport *
@@ -771,7 +772,7 @@
         extension = "so"
 
     # library name changed from libsingular to libSingular btw 3.x and 4.x
-    lib = os.environ['SAGE_LOCAL']+"/lib/libSingular."+extension
+    lib = "/usr/lib/%s/libsingular-Singular.so" % sysconfig.get_config_var('MULTIARCH')
 
     if not os.path.exists(lib):
         raise ImportError("cannot locate Singular library ({})".format(lib))
--- a/sage/src/sage/misc/cython.py
+++ b/sage/src/sage/misc/cython.py
@@ -304,10 +304,10 @@
     Before :trac:`12975`, it would have beeen needed to write ``#clang c++``,
     but upper case ``C++`` has resulted in an error::
 
-        sage: code = [
+        sage: import sysconfig; code = [
         ... "#clang C++",
-        ... "#cinclude %s/include/singular %s/include/factory"%(SAGE_LOCAL, SAGE_LOCAL),
-        ... "#clib m readline Singular givaro ntl gmpxx gmp",
+        ... "#cinclude /usr/include/singular /usr/include/singular/singular /usr/include/{0}/singular /usr/include/{0}/singular/singular".format(sysconfig.get_config_var('MULTIARCH')),
+        ... "#clib m readline singular-Singular givaro ntl gmpxx gmp",
         ... "from sage.rings.polynomial.multi_polynomial_libsingular cimport MPolynomial_libsingular",
         ... "from sage.libs.singular.polynomial cimport singular_polynomial_pow",
         ... "def test(MPolynomial_libsingular p):",
--- a/sage/src/sage/interfaces/tests.py
+++ b/sage/src/sage/interfaces/tests.py
@@ -39,7 +39,7 @@
     0
     sage: subprocess.call("echo syntax error | ipython", **kwds) in (0,1)
     True
-    sage: subprocess.call("echo syntax error | singular", **kwds)
+    sage: subprocess.call("echo syntax error | Singular", **kwds)
     0
 """
 from __future__ import print_function
